# **Analyzing Students' Mental Health**

Does going to university in a different country affect your mental health? A Japanese international university surveyed its students in 2018 and published a study the following year that was approved by several ethical and regulatory boards.

The study found that international students have a higher risk of mental health difficulties than the general population, and that social connectedness (belonging to a social group) and acculturative stress (stress associated with joining a new culture) are predictive of depression.

For this project, I will explore the \`students\` data using SQL and Python to find out if we can come to a similar conclusion for international students and see if the length of stay is a contributing factor.

# Overview of the dataset

In [ ]:
SELECT *
FROM [students].[dbo].[students.csv]

# Explore and understand the data




<span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">Count the number of records in the dataset as<i> <b>total_records</b></i>, and see how many records we have for each student type as <b><i>count_inter_dom</i></b>.</span>

In [ ]:
SELECT
	COUNT(*) AS total_records
FROM [students].[dbo].[students.csv]

(1 row affected)

Total execution time: 00:00:00.043

total_records 286

In [ ]:
SELECT
	inter_dom AS student_type,
	COUNT(*) AS count_inter_dom
FROM [students].[dbo].[students.csv]
GROUP BY inter_dom;

(3 rows affected)

Total execution time: 00:00:00.012

student_type count_inter_dom 18 Dom 67 Inter 201

# Filter to understand the data for each student type

<span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">Explore the data for each student type in <b><i>inter_dom</i></b></span> <span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">&nbsp;by doing three queries that filter for the two student types represented in the table, as well as the students with unknown status (</span>`NULL`<span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">).</span>

### _**International student**_

In [ ]:
-- Query for students with student type 'inter'
SELECT *
FROM [students].[dbo].[students.csv]
WHERE inter_dom = 'Inter';

(201 rows affected)

Total execution time: 00:00:00.021

inter_dom region gender academic age age_cate stay stay_cate japanese japanese_cate english english_cate intimate religion suicide dep deptype todep depsev tosc apd ahome aph afear acs aguilt amiscell toas partner friends parents relative profess phone doctor reli alone others internet partner_bi friends_bi parents_bi relative_bi professional_bi phone_bi doctor_bi religion_bi alone_bi others_bi internet_bi Inter SEA Male Grad 24 4 5 Long 3 Average 5 High Yes No No No 0 Min 34 23 9 11 8 11 2 27 91 5 5 6 3 2 1 4 1 3 4 Yes Yes Yes No No No No No No No No Inter SEA Male Grad 28 5 1 Short 4 High 4 High No No No No 2 Min 48 8 7 5 4 3 2 10 39 7 7 7 4 4 4 4 1 1 1 Yes Yes Yes No No No No No No No No Inter SEA Male Grad 25 4 6 Long 4 High 4 High Yes Yes No No No 2 Min 41 13 4 7 6 4 3 14 51 3 3 3 1 1 2 1 1 1 1 No No No No No No No No No No No Inter EA Female Grad 29 5 1 Short 2 Low 3 Average No No No No No 3 Min 37 16 10 10 8 6 4 21 75 5 5 5 5 5 2 2 2 4 4 Yes Yes Yes Yes Yes No No No No No No Inter EA Female Grad 28 5 1 Short 1 Low 3 Average Yes No No No No 3 Min 37 15 12 5 8 7 4 31 82 5 5 5 2 5 2 5 5 4 4 Yes Yes Yes No Yes No Yes Yes No No No Inter SEA Male Grad 24 4 6 Long 3 Average 4 High Yes No No No No 6 Mild 38 18 8 10 8 7 3 29 83 6 5 4 2 1 1 2 1 5 1 Yes Yes No No No No No No Yes No No Inter SA Male Grad 23 4 1 Short 3 Average 5 High Yes No No No No 3 Min 46 17 6 10 5 3 2 15 58 7 5 7 2 2 1 5 1 1 1 Yes Yes Yes No No No Yes No No No No Inter SEA Female Grad 30 5 2 Medium 1 Low 1 Low Yes Yes Yes No No 9 Mild 41 16 20 19 15 11 6 40 127 7 2 2 2 6 2 1 1 3 1 Yes No No No Yes No No No No No No Inter SEA Female Grad 25 4 4 Long 4 High 4 High No No No Yes Other 7 Mild 36 22 12 13 13 10 6 33 109 4 4 4 4 4 4 4 4 4 4 No No No No No No No No No No No Inter Others Male Grad 31 5 2 Medium 1 Low 4 High Yes Yes No No No 3 Min 48 8 4 5 12 3 2 17 51 1 1 1 2 1 1 2 1 2 1 No No No No No No No No No No No Inter Others Female Grad 28 5 1 Short 1 Low 2 Low No Yes No No No 5 Mild 32 24 8 10 8 6 6 30 92 7 3 7 7 3 1 3 1 3 5 Yes No Yes Yes No No No No No Yes No Inter SEA Female Grad 31 5 1 Short 1 Low 4 High Yes Yes No No No 8 Mild 47 17 12 14 13 9 4 26 95 6 2 6 2 2 2 2 4 2 5 Yes No Yes No No No No No No Yes No Inter SA Male Grad 29 5 1 Short 1 Low 4 High Yes Yes No No No 1 Min 48 8 11 5 4 7 2 17 54 7 5 7 7 1 1 1 4 2 1 Yes Yes Yes Yes No No No No No No No Inter EA Male Grad 23 4 1 Short 3 Average 4 High Yes Yes No No No 3 Min 32 9 8 5 4 6 7 18 57 7 5 3 2 2 2 2 2 1 1 Yes Yes No No No No No No No No No Inter SEA Female Grad 31 5 1 Short 1 Low 3 Average Yes No Yes No No 9 Mild 31 23 16 15 8 12 8 30 112 1 4 1 1 3 2 3 3 3 3 No No No No No No No No No No No Inter Others Female Grad 30 5 1 Short 1 Low 5 High Yes Yes No Yes Other 6 Mild 40 19 9 5 4 13 2 22 74 7 3 5 3 5 3 5 5 1 1 Yes No Yes No Yes No Yes Yes No No No Inter EA Female Grad 31 5 1 Short 1 Low 3 Average Yes No No No No 3 Min 48 11 13 7 4 8 5 15 63 6 5 2 2 1 1 1 1 3 1 Yes Yes No No No No No No No No No Inter Others Female Grad 29 5 1 Short 1 Low 5 High Yes Yes No No No 3 Min 48 16 4 5 4 3 2 13 47 7 5 7 3 1 1 5 1 1 1 Yes Yes Yes No No No Yes No No No No Inter SEA Female Under 19 2 1 Short 3 Average 5 High No No No No 7 Mild 44 11 8 5 7 4 2 18 55 5 3 6 7 5 2 2 1 1 1 Yes No Yes Yes Yes No No No No No No Inter SEA Male Under 25 4 3 Medium 2 Low 4 High Yes No No No 1 Min 36 13 10 7 5 3 3 19 60 1 5 1 1 1 1 1 1 6 1 No Yes No No No No No No Yes No No Inter SEA Male Under 18 1 1 Short 1 Low 4 High No No No No 4 Min 26 18 4 7 4 4 2 27 66 6 4 6 6 6 3 4 3 6 5 Yes No Yes Yes Yes No No No Yes Yes No Inter SEA Male Under 18 1 1 Short 2 Low 3 Average No No No No 3 Min 26 11 5 5 7 3 2 33 66 2 2 2 2 2 2 2 1 1 2 No No No No No No No No No No No Inter SEA Male Under 19 2 1 Short 1 Low 4 High Yes Yes Yes Other 13 Mod 25 25 16 10 11 6 4 22 94 1 2 6 1 1 1 1 1 7 1 No No Yes No No No No No Yes No No Inter SEA Male Under 20 2 3 Medium 3 Average 4 High No No No No 1 Min 34 24 16 15 12 11 8 37 123 6 7 7 6 6 6 5 6 5 5 Yes Yes Yes Yes Ye

### **_Domestic student_**

In [ ]:
-- Query for students with student type 'dom'
SELECT *
FROM [students].[dbo].[students.csv]
WHERE inter_dom = 'Dom';

(67 rows affected)

Total execution time: 00:00:00.025

inter_dom region gender academic age age_cate stay stay_cate japanese japanese_cate english english_cate intimate religion suicide dep deptype todep depsev tosc apd ahome aph afear acs aguilt amiscell toas partner friends parents relative profess phone doctor reli alone others internet partner_bi friends_bi parents_bi relative_bi professional_bi phone_bi doctor_bi religion_bi alone_bi others_bi internet_bi Dom JAP Female Grad 27 5 2 Medium 3 Average 3 Average Yes Yes No Yes Major 12 Mod 47 16 11 5 8 7 3 31 81 7 3 7 1 6 6 1 5 4 1 Yes No Yes No Yes Yes No Yes No No No Dom JAP Female Under 18 1 1 Short 5 High 3 Average No No No No No 9 Mild 48 9 4 5 4 3 2 10 37 4 4 4 4 1 1 1 1 1 1 4 No No No No No No No No No No No Dom JAP Female Under 21 3 3 Medium 5 High 3 Average Yes No No No No 7 Mild 40 16 8 10 8 6 4 20 72 6 6 7 1 1 1 5 1 1 1 4 Yes Yes Yes No No No Yes No No No No Dom JAP Male Under 20 2 3 Medium 5 High 1 Low No No No No No 3 Min 47 11 4 5 4 5 2 12 43 1 5 5 3 1 1 3 1 1 1 3 No Yes Yes No No No No No No No No Dom JAP Female Under 21 3 3 Medium 5 High 1 Low No No Yes Yes Other 10 Mod 48 8 4 5 4 3 2 10 36 7 5 7 1 1 1 1 1 1 1 1 Yes Yes Yes No No No No No No No No Dom JAP Male Under 20 2 3 Medium 5 High 3 Average No No Yes No No 6 Mild 34 13 6 7 8 4 3 15 56 1 2 1 1 2 2 2 1 1 1 1 No No No No No No No No No No No Dom JAP Female Under 18 1 1 Short 5 High 3 Average No No No No No 6 Mild 42 14 6 10 4 5 4 28 71 6 4 6 4 4 5 5 3 2 2 2 Yes No Yes No No Yes Yes No No No No Dom JAP Female Under 22 3 4 Long 5 High 3 Average No No No No No 0 Min 48 8 4 5 4 3 2 10 36 5 7 7 5 5 5 5 1 1 3 3 Yes Yes Yes Yes Yes Yes Yes No No No No Dom JAP Female Under 22 3 4 Long 5 High 1 Low No Yes Yes Yes Major 13 Mod 24 17 15 11 11 7 6 29 96 2 2 2 2 2 2 2 2 5 2 2 No No No No No No No No Yes No No Dom JAP Male Under 20 2 3 Medium 5 High 3 Average Yes No No Yes Other 9 Mild 48 8 4 5 4 3 2 11 37 7 6 3 1 1 1 1 1 1 1 1 Yes Yes No No No No No No No No No Dom JAP Female Under 21 3 3 Medium 5 High 4 High No Yes No No No 7 Mild 30 19 7 12 8 10 6 28 90 5 5 5 3 2 2 2 3 2 2 2 Yes Yes Yes No No No No No No No No Dom JAP Female Under 21 3 3 Medium 5 High 3 Average Yes No No Yes Major 20 Sev 34 11 5 7 4 9 6 33 75 7 3 1 1 1 1 1 1 6 1 1 Yes No No No No No No No Yes No No Dom JAP Female Under 20 2 3 Medium 5 High 2 Low Yes No Yes Yes Major 21 Sev 14 17 11 10 10 6 4 19 77 1 4 7 6 2 2 2 1 1 2 4 No No Yes Yes No No No No No No No Dom JAP Male Under 21 3 2 Medium 5 High 3 Average No No Yes Yes Major 23 Sev 8 10 5 5 4 3 3 13 43 1 1 2 1 6 4 5 2 7 4 4 No No No No Yes No Yes No Yes No No Dom JAP Female Under 21 3 3 Medium 5 High 3 Average No No No No No 2 Min 48 8 13 5 4 3 3 16 52 4 6 6 2 1 1 1 1 1 1 4 No Yes Yes No No No No No No No No Dom JAP Male Under 20 2 2 Medium 5 High 3 Average No No No No No 13 Mod 47 8 12 5 4 10 7 20 66 5 5 7 5 4 4 4 2 1 4 1 Yes Yes Yes Yes No No No No No No No Dom JAP Male Under 21 3 2 Medium 5 High 3 Average Yes No No Yes Other 7 Mild 48 14 4 10 7 8 3 19 65 7 5 5 2 3 3 3 1 1 1 1 Yes Yes Yes No No No No No No No No Dom JAP Male Under 21 3 4 Long 2 Low 5 High Yes No Yes No No 11 Mod 24 27 8 14 12 10 4 21 96 4 4 4 4 4 4 4 4 6 1 4 No No No No No No No No Yes No No Dom JAP Female Under 22 3 1 Short 5 High 5 High Yes No No No No 5 Mild 28 16 4 5 4 3 2 10 44 6 2 1 1 1 1 1 1 6 1 1 Yes No No No No No No No Yes No No Dom JAP Female Under 21 3 3 Medium 5 High 3 Average No No No No No 1 Min 41 8 4 10 7 3 2 15 49 7 4 7 4 6 2 6 1 2 1 3 Yes No Yes No Yes No Yes No No No No Dom JAP Male Under 21 3 1 Short 4 High 3 Average Yes Yes No No No 8 Mild 48 8 4 5 4 3 2 10 36 7 7 7 7 4 1 2 1 1 1 1 Yes Yes Yes Yes No No No No No No No Dom JAP Female Under 19 2 2 Medium 5 High 3 Average No No No No No 8 Mild 39 8 5 10 5 4 4 24 60 3 5 3 1 1 1 1 1 3 2 3 No Yes No No No No No No No No No Dom JAP Male Under 23 4 4 Long 5 High 3 Average No Yes No No No 11 Mod 38 17 8 10 8 6 4 21 74 7 7 7 5 4 4 4 3 2 2 3 Yes Yes Yes Yes No No No No No No No Dom JAP Female Under 18 1 1 Short 5 High 3 Average 

### **_Unknown student_**

In [ ]:
-- Query for students with unknown student type
SELECT *
FROM [students].[dbo].[students.csv]
WHERE inter_dom IS NULL;

(0 rows affected)

Total execution time: 00:00:00.008

inter_dom region gender academic age age_cate stay stay_cate japanese japanese_cate english english_cate intimate religion suicide dep deptype todep depsev tosc apd ahome aph afear acs aguilt amiscell toas partner friends parents relative profess phone doctor reli alone others internet partner_bi friends_bi parents_bi relative_bi professional_bi phone_bi doctor_bi religion_bi alone_bi others_bi internet_bi

# Query the summary statistics of the diagnostics scores for all students

<span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">Find the summary statistics for each diagnostic test using aggregate functions. Round the averages to two decimal places and use aliases <i><b>min_phq</b></i></span><span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">, <i><b>max_phq</b></i>, and <b><i>avg_phq</i></b></span><span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">; repeat this structure for all tests (<i><b>_scs</b></i></span> <span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">&nbsp;and <i><b>_as</b></i></span><span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">).</span>

In [ ]:
SELECT
-- Query for calculating scores of Depression of all students
	ROUND(AVG(CAST(todep AS int)),2) AS avg_phq,
    ROUND(MIN(CAST(todep AS int)),2) AS min_phq,
	ROUND(MAX(CAST(todep AS int)),2) AS max_phq,

-- Query for calculating scores of Social Connectedness of all students
    ROUND(AVG(CAST(tosc AS int)),2) AS avg_scs,
	ROUND(MIN(CAST(tosc AS int)),2) AS min_scs,
	ROUND(MAX(CAST(tosc AS int)),2) AS max_scs,

-- Query for calculating score of Acculturative Stress of all students
    ROUND(AVG(CAST(toas AS int)),2) AS avg_as,
	ROUND(MIN(CAST(toas AS int)),2) AS min_as,
	ROUND(MAX(CAST(toas AS int)),2) AS max_as

FROM [students].[dbo].[students.csv];

(1 row affected)

Total execution time: 00:00:00.006

avg_phq min_phq max_phq avg_scs min_scs max_scs avg_as min_as max_as 7 0 25 35 0 48 67 0 145

# Summarize the data for international students only

<span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">Narrow down the results down further to see the summary statistics for international students only through filtering and grouping.</span>

In [ ]:
SELECT
-- Query for calculating average Depression score of international students
	ROUND(AVG(CAST(todep AS int)),2) AS inter_avg_phq,
	ROUND(MIN(CAST(todep AS int)),2) AS inter_min_phq,
	ROUND(MAX(CAST(todep AS int)),2) AS inter_max_phq,

-- Query for calculating Social Connectedness scores of international students
	ROUND(AVG(CAST(tosc AS int)),2) AS inter_avg_scs,
	ROUND(MIN(CAST(tosc AS int)),2) AS inter_min_scs,
	ROUND(MAX(CAST(tosc AS int)),2) AS inter_max_scs,

-- Query for calculating Acculturative Stress scores of international students
	ROUND(AVG(CAST(toas AS int)),2) AS inter_avg_as,
	ROUND(MIN(CAST(toas AS int)),2) AS inter_min_as,
	ROUND(MAX(CAST(toas AS int)),2) AS inter_max_as
FROM [students].[dbo].[students.csv]
WHERE inter_dom = 'Inter';

(1 row affected)

Total execution time: 00:00:00.012

inter_avg_phq inter_min_phq inter_max_phq inter_avg_scs inter_min_scs inter_max_scs inter_avg_as inter_min_as inter_max_as 8 0 25 37 11 48 75 36 145

# See the impact of length of stay

<span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">Let's see how the length of stay of an international student impacts the&nbsp;</span> <span style="box-sizing: inherit; font-weight: 700; color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">average</span> <span style="color: rgb(101, 112, 124); font-family: Studio-Feixen-Sans, Arial, sans-serif; background-color: rgb(255, 255, 255);">&nbsp;diagnostic scores, I order the results by descending order of the length of stay.</span>

In [ ]:
-- Find the average scores by length of stay for international students, and view them in descending order
SELECT CAST(stay AS numeric) AS length_of_stay,
       ROUND(AVG(CAST(todep AS int)), 2) AS average_phq,
       ROUND(AVG(CAST(tosc AS int)), 2) AS average_scs,
       ROUND(AVG(CAST(toas AS int)), 2) AS average_as
FROM [students].[dbo].[students.csv]
WHERE inter_dom = 'Inter'
GROUP BY CAST(stay AS numeric)
ORDER BY length_of_stay DESC;

(9 rows affected)

Total execution time: 00:00:00.069

length_of_stay average_phq average_scs average_as 10 13 32 50 8 10 44 65 7 4 48 45 6 6 38 58 5 0 34 91 4 8 33 87 3 9 37 78 2 8 37 77 1 7 38 72

# Analyze the correlation between length of stay and diagnostic scores

With the result from section 5, I use Python to calculate the correlation between length of stay and average diagnostic scores for international students.

In [ ]:
# Calculating correlation coefficients

import pyodbc
import pandas as pd
data = [
[10,13,32,50],
[8,10,44,65],
[7,4,48,45],
[6,6,38,58],
[5,0,34,91],
[4,8,33,87],
[3,9,37,78],
[2,8,37,77],
[1,7,38,72]
]
stay_depression = pd.DataFrame(data, columns=['length_of_stay', 'average_phq','average_scs','average_as'])
print(stay_depression.corr())

                length_of_stay  average_phq  average_scs  average_as
length_of_stay        1.000000     0.273703     0.148855   -0.636160
average_phq           0.273703     1.000000    -0.181039   -0.324932
average_scs           0.148855    -0.181039     1.000000   -0.521304
average_as           -0.636160    -0.324932    -0.521304    1.000000


# CONCLUSION

In conclusion, even though the study found that social connectedness and acculturative stress are predictive of depression for international students, further exploration into the dataset shows that the length of stay is not quite a contributing factor.

The correlation coefficient between "stay" with "average\_phq" and "average\_scs" is 0.27 and 0.14 respectively. The numbers show weakly correlated relationships between length of stay with depression and social connectedness, which means two things:

- The longer a student stays in a new country doesn't neccessarily lead to the more they feel belonged to the native social group;
- And level of depressionn also doesn't strongly depend on how long they have been living in a new country.

On the other hand, the correlation coefficient between "stay" and "average\_as" is -0.64 indicates a quite strongly negative relationship, which means the longer the stay associated with the less stress associated with joining a new culture as international students will gradually accomodate themselves to the new environment.